kad-dht/README.md

```diff
@@ -268,23 +268,55 @@ heuristic of the value to make the decision.
 
 ### Content provider advertisement and discovery
 
-Nodes must keep track of which nodes advertise that they provide a given key
-(CID). These provider advertisements should expire, by default, after 24 hours.
-These records are managed through the `ADD_PROVIDER` and `GET_PROVIDERS`
+There are two things at play with regard to provider record (and therefore content)
+liveness and reachability:
+
+Content providers need to make sure that their content is reachable, despite peer churn
+and nodes that store and serve provider records need to make sure that the Multihashes whose 
+records they store are still served by the content provider.
+
+The following two parameters help cover both of these cases.
+1. **Provider Record Republish Interval (24hrs):** The content provider 
+needs to make sure that the nodes chosen to store the provider record 
+remain online when clients ask for the record. In order to 
+guarantee this, while taking into account the peer churn, content providers
+republish the records they want to provide every 24 hours.
+2. **Provider Record Expiration Interval (48hrs):** The network needs to provide
```

Comment:
@yiannisbot while we're changing the times any reason not to do this too? Seems like it'd be reasonable since expiration is based on the same properties. It'd also make things easier to reason about.

Reply:
I agree with changing them together. However, unless I am missing something in the code, they will have to be two separate PRs:
1. Update `kubo`'s default `Reprovider.Interval` [here](https://github.com/cortze/kubo/blob/0e0136100ed8aaea6c27c89c9b73471f542d8c1d/config/init.go#L78)
2. Update `go-libp2p-kad-dht`'s `ProvideValidity` [here](https://github.com/cortze/go-libp2p-kad-dht/blob/6a75b5910b5d6a60e17d6d69524d11783a57ad6b/providers/providers_manager.go#L29)

Reply:
> You mean in the PR

No, I mean to change the expiration time for `PUT_VALUE` records in addition to `ADD_PROVIDER` records.

i.e. for the IPFS Public DHT the expiration time for `PUT_VALUE` (i.e. IPNS and the deprecated public key records) is 36hrs
https://github.com/libp2p/go-libp2p-kad-dht/blob/dae5a9a5bd9c7cc8cfb5073c711bc308efad0ada/internal/config/config.go#L117. It seems like this could be 48hrs as well rather than 36.